packages/devtools_app/lib/src/shared/memory/heap_object.dart

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'class_name.dart';
+import 'heap_data.dart';
+import 'simple_items.dart';
+
+@immutable
+class HeapObject {
+  const HeapObject(this.heap, {required this.index});
+
+  final HeapData heap;
+
+  /// If object is null, it exists in live app, but is not
+  /// located in heap.
+  final int? index;
+
+  Iterable<int>? _refs(RefDirection direction) {
+    final theObject = index;
```

Comment:
why are object and index used as interchangeable language in this class? this should be explained or the names should be made the same for consistency